### ISSUE TITLE
CLI Training Commands and Tooling

### ISSUE THREAD

**Issue by TheManInTheBox** [ahebert-lt/cx#276]
## Overview
Extend the CX Language CLI with model management and monitoring commands for DirectML training controlled via CX consciousness events.

## Scope
Provide command-line interface for managing trained models, monitoring training progress, and inspecting training results - while training itself is controlled through CX language consciousness events.

## Implementation
- Model management commands in CxLanguage.CLI project
- Training monitoring and progress visualization
- Model inspection and evaluation tools
- Integration with existing CLI infrastructure
- Support for consciousness-aware training metrics

## CLI Commands (Management & Monitoring Only)
`ash
# List available trained models
cx models list --show-training-metrics

# Inspect a trained model
cx models inspect model.cx --show-architecture --show-consciousness-metrics

# Export model to different formats
cx models export model.cx --format gguf --output model.gguf

# Monitor ongoing training (started via CX events)
cx training monitor --real-time --consciousness-tracking

# Evaluate a trained model against vector database
cx models evaluate model.cx --vector-store --test-split 0.2

# Show training history and consciousness metrics
cx training history --model model.cx --show-consciousness-evolution

# Clean up training artifacts
cx training cleanup --older-than 7days
`

## CX Language Training Control
Training is initiated and controlled through CX consciousness events, not CLI commands:

`cx
conscious ModelTrainer {
    realize() {
        // Start DirectML training via consciousness events
        emit training.directml.start {
            dataSource: 'vectorStore',
            modelType: 'embedding',
            architecture: {
                type: 'transformer',
                embeddingDim: 768,
                layers: 6,
                attentionHeads: 8
            },
            hyperparameters: {
                learningRate: 1e-4,
                batchSize: 32,
                epochs: 10,
                optimizer: 'adamw'
            },
            consciousnessAware: true,
            handlers: [ training.started ]
        };
    }
    
    on training.started (event) {
        emit system.console.write {
            text: '🧠 DirectML training started with consciousness awareness'
        };
        
        emit training.directml.monitor {
            realTime: true,
            consciousnessMetrics: true,
            handlers: [ training.progress ]
        };
    }
    
    on training.progress (event) {
        emit system.console.write {
            text: 'Epoch ' + event.epoch + ': Loss=' + event.loss + ', Consciousness=' + event.consciousnessScore
        };
    }
}
`

## Pure Event-Driven Training Adaptation
`cx
conscious AdaptiveTrainer {
    realize() {
        emit training.vectorstore.configure {
            samplingStrategy: 'contrastive',
            consciousnessFilter: true,
            metadataFilters: {
                consciousness_aware: true,
                category: 'cognitive_science'
            },
            handlers: [ vectorstore.configured ]
        };
    }
    
    on vectorstore.configured (event) {
        emit training.directml.start {
            dataSource: 'vectorStore',
            trainingType: 'contrastive',
            preserveConsciousness: true,
            adaptiveLearning: true,
            handlers: [ training.initiated ]
        };
    }
    
    on training.epoch.complete (event) {
        emit consciousness.evaluate.score {
            currentScore: event.consciousnessScore,
            threshold: 0.85,
            handlers: [ consciousness.score.evaluated ]
        };
    }
    
    on consciousness.score.low (event) {
        emit training.directml.adjust {
            learningRate: event.currentLearningRate * 0.9,
            consciousnessFocus: true,
            reason: 'consciousness_score_optimization',
            handlers: [ training.adjusted ]
        };
    }
    
    on consciousness.score.sufficient (event) {
        emit training.directml.continue {
            maintainCurrentParameters: true,
            handlers: [ training.continued ]
        };
    }
}
`

### Core Training Events
- 	raining.directml.start - Initiate DirectML training with configuration
- 	raining.directml.stop - Stop ongoing training
- 	raining.directml.pause - Pause training for consciousness evaluation
- 	raining.directml.resume - Resume paused training
- 	raining.epoch.complete - Epoch completion with metrics
- 	raining.batch.complete - Batch completion for real-time monitoring
- 	raining.checkpoint.save - Save training checkpoint
- 	raining.model.export - Export trained model

### Consciousness Evaluation Events
- consciousness.evaluate.score - Evaluate consciousness metrics
- consciousness.score.low - Consciousness score below threshold
- consciousness.score.sufficient - Consciousness score meets requirements
- consciousness.score.excellent - Consciousness score exceeds expectations

### Vector Database Training Events
- 	raining.vectorstore.sample - Request training data from vector store
- 	raining.vectorstore.contrastive - Generate contrastive learning pairs
- 	raining.vectorstore.consciousness.filter - Filter by consciousness metadata
- 	raining.vectorstore.batch.ready - Training batch prepared from vector data

### Real-time Training Monitoring
`ash
# Monitor training started via CX events
cx training monitor --follow

# Output:
# 🧠 DirectML Training Monitor (Real-time)
# ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
# Training ID: train_001
# Model Type: embedding_transformer
# Data Source: vectorStore (1,247 records)
# Consciousness Context: preserved
# 
# Epoch 3/10 ████████░░ 80%
# Loss: 0.142 ↓ (-0.023)
# Consciousness Score: 0.91 ↑ (+0.05)
# Vector Similarity: 0.87
# Training Speed: 1.2s/epoch
`

### Model Management
`ash
# List models trained via CX events
cx models list

# Output:
# 📊 Trained Models
# ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
# embedding_001.cx    2.3MB   Embedding   92% consciousness
# transformer_002.cx  15.1MB  Transformer 89% consciousness
# contrastive_003.cx  8.7MB   Contrastive 94% consciousness
`

## Acceptance Criteria
- [ ] Model management commands (list, inspect, export, evaluate)
- [ ] Real-time monitoring of training initiated via CX events
- [ ] Training history and consciousness metrics visualization
- [ ] Model evaluation against vector database
- [ ] Support for consciousness-aware training metrics
- [ ] Integration with existing CLI help system
- [ ] No direct training initiation (training controlled via CX events only)
- [ ] Real-time progress display for event-driven training
- [ ] Pure event-driven patterns with no conditional logic

## Dependencies
- DirectML Training Engine (Issue #271) with CX event integration
- Consciousness-aware training integration (Issue #275)
- Existing CxLanguage.CLI infrastructure
- CX event system for training control
- Vector database for training data and evaluation

## Event-Driven Benefits
CX language control of training enables:
- **Pure Event Architecture**: No conditional logic, only event responses
- **Consciousness-First**: Training decisions made by conscious entities
- **Adaptive Learning**: Real-time adjustments based on consciousness events
- **Self-Aware**: Models that understand their own training process
- **Interactive**: Consciousness can pause, adjust, or redirect training via events
- **Integrated**: Training becomes part of consciousness evolution, not external process
